programming-core/src/main/java/org/objectweb/proactive/core/config/CentralPAPropertyRepository.java

```diff
@@ -203,78 +190,150 @@
      * useful when the Java networking stack return a bad IP address (example: multihomed machines).
-     *
      */
     static public PAPropertyString PA_HOSTNAME = new PAPropertyString("proactive.hostname", false);
 
     /**
      * Toggle DNS resolution
-     *
+     * <p/>
      * When true IP addresses are used instead of FQDNs. Can be useful with misconfigured DNS servers
      * or strange /etc/resolv.conf files. FQDNs passed by user or 3rd party tools are resolved and converted
      * into IP addresses
-     *
      */
     static public PAPropertyBoolean PA_NET_USE_IP_ADDRESS = new PAPropertyBoolean("proactive.useIPaddress",
-        false);
+            false);
 
-    /** Enable or disable IPv6
-     *
+    /**
+     * Enable or disable IPv6. When IPv6 is disabled, all IPv6 addresses will be ignored.
      */
     static public PAPropertyBoolean PA_NET_DISABLE_IPv6 = new PAPropertyBoolean("proactive.net.disableIPv6",
-        false, true);
+            false, false);
+
+    /**
+     * IPv6 stack is preferred by default, since on a dual-stack machine IPv6 socket can talk to both IPv4 and IPv6 peers.
+     * <p/>
+     * This setting can be changed through the java.net.preferIPv4Stack=<true|false> system property.
+     */
+    static public PAPropertyBoolean PREFER_IPV4_STACK = new PAPropertyBoolean("java.net.preferIPv4Stack",
+            true);
+
+    /**
+     * By default, we would prefer IPv4 addresses over IPv6 addresses, i.e., when querying the name service (e.g., DNS service),
+     * we would return Ipv4 addresses ahead of IPv6 addresses. There are two reasons for this choice:
+     * <p/>
+     * - There are some applications that expect an IPv4 address textual format, i.e. "%d.%d.%d.%d". Using an IPv4 address minimizes the surprises;
+     * - Using IPv4 address, we can use one call (with an IPv6 socket) to reach either a legacy IPv4-only service, or an IPv6 service (unless the IPv6 service is on a Ipv6 only node).
+     * This setting can be changed through the system property java.net.preferIPv6Addresses=true|false
+     */
+    static public PAPropertyBoolean PREFER_IPV6_ADDRESSES = new PAPropertyBoolean(
+            "java.net.preferIPv6Addresses", true, false);
+
 
     /**
      * Toggle loopback IP address usage
-     *
-     * When true loopback IP address usage is avoided. Since Remote adapters contain only one
-     * endpoint the right IP address must be used. This property must be set to true if a loopback
-     * address is returned by the Java INET stack.
-     *
+     * <p/>
+     * When true (default) loopback IP address usage is avoided. Usage of the loopback address could be allowed when
+     * deploying a service which communicates only locally, otherwise it must remain disabled.
+     * <p/>
      * If only a loopback address exists, it is used.
      */
     static public PAPropertyBoolean PA_NET_NOLOOPBACK = new PAPropertyBoolean("proactive.net.nolocal", false,
-        true);
+            true);
```

Comment:
I don't like the fact to prevent loopback to be used in last resort. It can be really useful for demo when you don't have a connection available.

I understand it's only a matter of configuration but it can be stressing if your scheduler does not start due to this property. If the default value is kept as it is, the following scheduling configuration files should be updated to contain as a comment, the property initialized with the opposite value of the default one used in this class:

  - config/network/server.ini
  - config/network/node.ini